ambuda/views/auth.py

```diff
@@ -79,51 +91,99 @@ def _is_valid_reset_token(row: db.PasswordResetToken, raw_token: str, now=None):
     return True
 
 
+# Copied from https://wtforms.readthedocs.io/en/2.3.x/validators/
+class FieldLength(object):
+    def __init__(self, min=-1, max=-1, message=None):
+        self.min = min
+        self.max = max
+        if not message:
+            message = "Field must be between %i and %i characters long." % (min, max)
+        self.message = message
+
+    def __call__(self, form, field):
+        field_len = field.data and len(field.data) or 0
+        if field_len < self.min or self.max != -1 and field_len > self.max:
+            raise val.ValidationError(self.message)
+
+
+def get_field_validators(field_name: str, min_len: int, max_len: int):
+    return [
+        val.DataRequired(),
+        FieldLength(
+            min=min_len,
+            max=max_len,
+            message=f"{field_name.capitalize()} must be between {min_len} and {max_len} characters long",
+        ),
+    ]
+
+
+def get_username_validators(is_legacy: bool = False):
```

Comment:
Since the function is so short, let's split this into 2 separate functions instead, maybe `get_username_validators` and `get_legacy_username_validators`